src/components/register/RegisterChallengePayment.vue

```diff
@@ -0,0 +1,450 @@
+ *
+ * @example
+ * <register-challenge-payment />
+ *
+ * @see [Figma Design](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=6485-30201&t=5JpU0wrURbmXLMCm-1)
+ */
+
+// libraries
+import { colors } from 'quasar';
+import { computed, defineComponent, reactive, ref, watch } from 'vue';
+
+// composables
+import { i18n } from '../../boot/i18n';
+import { useFormatPrice } from '../../composables/useFormatPrice';
+
+// components
+import FormFieldCompany from '../global/FormFieldCompany.vue';
+import FormFieldDonation from '../form/FormFieldDonation.vue';
+import FormFieldPhone from '../global/FormFieldPhone.vue';
+import FormFieldRadioRequired from '../form/FormFieldRadioRequired.vue';
+import FormFieldSliderNumber from '../form/FormFieldSliderNumber.vue';
+import FormFieldTextRequired from '../global/FormFieldTextRequired.vue';
+import FormFieldVoucher from '../form/FormFieldVoucher.vue';
+
+// config
+import { rideToWorkByBikeConfig } from '../../boot/global_vars';
+
+// enums
+enum PaymentSubject {
+  individual = 'individual',
+  voucher = 'voucher',
+  company = 'company',
+  school = 'school',
+}
+
+// types
+import type { FormOption, FormPaymentVoucher } from '../types/Form';
+
+export default defineComponent({
+  name: 'RegisterChallengePayment',
+  components: {
+    FormFieldCompany,
+    FormFieldDonation,
+    FormFieldPhone,
+    FormFieldRadioRequired,
+    FormFieldSliderNumber,
+    FormFieldTextRequired,
+    FormFieldVoucher,
+  },
+  setup() {
+    // constants
+    const defaultPaymentAmountMax = Number(
+      rideToWorkByBikeConfig.entryFeePaymentMax,
+    );
+    const defaultPaymentAmountMin = Number(
+      rideToWorkByBikeConfig.entryFeePaymentMin,
+    );
+    const borderRadius = rideToWorkByBikeConfig.borderRadiusCardSmall;
+    const { getPaletteColor, lighten } = colors;
+    const primaryColor = getPaletteColor('primary');
+    const primaryLightColor = lighten(primaryColor, 90);
+
+    const optionsPaymentSubject: FormOption[] = reactive([
+      {
+        label: i18n.global.t(
+          'register.challenge.labelPaymentSubjectIndividual',
+        ),
+        value: PaymentSubject.individual,
+      },
+      {
+        label: i18n.global.t('register.challenge.labelPaymentSubjectVoucher'),
+        value: PaymentSubject.voucher,
+      },
+      {
+        label: i18n.global.t('register.challenge.labelPaymentSubjectCompany'),
+        value: PaymentSubject.company,
+      },
+      {
+        label: i18n.global.t('register.challenge.labelPaymentSubjectSchool'),
+        value: PaymentSubject.school,
+      },
+    ]);
+
+    const { formatPriceCurrency } = useFormatPrice();
+    const defaultPaymentOption = {
+      label: formatPriceCurrency(defaultPaymentAmountMin, 'CZK'),
+      value: String(defaultPaymentAmountMin),
+    };
+    const optionsPaymentAmount: FormOption[] = reactive([
+      defaultPaymentOption,
+      {
+        label: '500 Kč',
```

Comment:
You can use global config var for currency example: `currency = "CZK"` and you can use same global config var as `formatPriceCurrency()` func argument.

Reply:
Patch for label key value currency (use trans string):

```ts
label: `500 ${$('currencyUnitCzk')}`
```

Reply:
I approached it differently `formatPriceCurrency` uses JS internationalization API so it automatically converts to the correct currency label based on given currency.
I refactored the currency string 'CZK' into an Enum.